compiler/rustc_typeck/src/collect.rs

```diff
@@ -2683,6 +2683,14 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
         codegen_fn_attrs.flags |= CodegenFnAttrFlags::TRACK_CALLER;
     }
 
+    // With -Z panic-in-drop=abort, drop_in_place never unwinds.
+    if tcx.sess.opts.debugging_opts.panic_in_drop == PanicStrategy::Abort {
+        let drop_in_place = tcx.require_lang_item(LangItem::DropInPlace, None);
```

Comment:
You can check `if Some(id) == tcx.lang_items.drop_in_place()` instead of requiring the lang item.